### DISCUSSION TITLE
What is the recommended way to allow non-admin users to create a content item from the front end?

### DISCUSSION THREAD

**Discussion by DrewBrasher** [OrchardCMS/OrchardCore#13915] [Q&A]
For example, I'm planning to build a module to handle helpdesk requests. If I create a content type called `HelpRequest` and create an editor template for the admin side in Razor, what is the best way to let users submit a help request which would create a `HelpRequest` content item? Would I use the same editor template I create for the admin side on the front end somehow or would I just make a regular .net core razor form that on post would create the content type like mentioned here in https://github.com/OrchardCMS/OrchardCore/discussions/8694 ?

**Comment by hishamco** [comment]
You could create a new content item programmatically from the front end, I already did something similar last two weeks or so

```csharp
    var request = await _contentManager.NewAsync("HelpRequest");

    request.DisplayText = "Request # 123";
    // Set the values of the fields
    request.Content.HelpRequest.Description.Text = "Description goes here ...";
    ...

    await _contentManager.CreateAsync(request);
```

**Reply by jtkech** [reply]
After creating a new item you have to use helpers to mutate parts/fields, from memory like this

            var somePart = contentItem.As<SomePart>();

Or maybe better

            var somePart = contentItem.GetOrCreate<SomePart>();

Then mutate the part fields and apply it to the item

            contentItem.Apply(SomePart);

Finally

    var result = await _contentManager.UpdateValidateAndCreateAsync(contentItem, VersionOptions.Published);
    if (!result.Succeeded)
    {
        ...
    }

**Reply by DrewBrasher** [reply]
If the content type has a media field that is using the attached editor on the admin side, how could I attach a file uploaded on the front end?

**Comment by hishamco** [comment]
> If the content type has a media field that is using the attached editor on the admin side, how could I attach a file uploaded on the front end?

@DrewBrasher  you could use `IMediaFileStore` to create your media, don't forget to set the value for the media field

**Reply by jtkech** [reply]
For attached medias look at how the `MediaFieldDisplayDriver` uses the `AttachedMediaFieldFileService`.

**Reply by DrewBrasher** [reply]
Thanks, I will take a look at that. 
Is it possible to use the the built in editor ([MediaField-Attached.Edit.cshtml](https://github.com/OrchardCMS/OrchardCore/blob/c820a040ef8c62e426162fdf53ddd8c0697f4901/src/OrchardCore.Modules/OrchardCore.Media/Views/MediaField-Attached.Edit.cshtml)) on the front end? I pretty much need the same functionality of the admin side on the front end.

**Reply by hishamco** [reply]
I'm not sure what's wrong with using `IMediaFileStore` with a simple upload input?!
